.github/workflows/deploy.yml

```diff
@@ -27,17 +27,33 @@ jobs:
   deploy:
     name: Deploy
     runs-on: ubuntu-latest
+    container:
+        image: ubuntu:latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: ruby/setup-ruby@v1
+      - name: Update packages
+        run: |
+            apt-get update -qq
+      - name: Install dependencies
+        run: |
+            apt-get install -qq -y jq rsync git libyaml-0-2 npm
+      - name: Checkout git repository
+        uses: actions/checkout@v3
+      - name: Set GitHub workspace as git safe.directory # Required to work around: https://github.com/actions/checkout/issues/766
+        run: |
+            git config --global --add safe.directory "$GITHUB_WORKSPACE"
+      - name: Install Ruby
+        uses: ruby/setup-ruby@v1
         with:
-          bundler-cache: true
-      - uses: actions/setup-node@v2
+            bundler-cache: true
+        env:
+            ImageOS: ubuntu22
+      - name: Install Node.js
+        uses: actions/setup-node@v2
```

Comment:
```suggestion
        uses: actions/setup-node@v3
```

Reply:
You are absolutely right. Thank you for catching this @kou!

It looks like the [version of Node.js in the Ubuntu 22.04 package repositories is version `12.22.9`](https://packages.ubuntu.com/jammy/web/nodejs). I had falsely assumed it would be version 16 since that was the version used by the `ubuntu-latest` runner prior to the breaking upgrade to Node 18.

I should have looked at the build log more closely to confirm this. My apologies for this oversight.

Moving forward:

1. I think we should explicitly specify the version of Node.js we want to use to avoid any ambiguity (this is recommended in the `actions/setup-node` documentation).
2. I agree that we should use `v3` of `actions/setup-node`.
3. Updating Webpack with `npm install webpack` seems like a good idea, too. However, this may require us to specify the `output.hashFunction` in the Webpack config too. I'll investigate this further.

Reply:
> Updating Webpack with `npm install webpack` seems like a good idea, too. However, this may require us to specify the `output.hashFunction` in the Webpack config too. I'll investigate this further.

I didn't try it but it may not be needed because the latest webpack include fallback `md4` implementation as you said.

Reply:
For explicitly specifying the Node.js version to use, I think a reasonable approach would be to update the `.nvmrc` file to use `lts/hydrogen` (which should resolve to the current Node.js LTS version `18.14.2`).

The `.nvmrc` file can then be used with `actions/setup-node` with the [`node-version-file` input](https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#node-version-file).

See: https://github.com/nvm-sh/nvm#long-term-support and https://github.com/nodejs/Release#release-schedule.

Reply:
It makes sense.
We may be able to `lts/*`. If we get similar error again, we'll be able to resolve by just `npm install XXX` (use the latest XXX) because most packages will support the latest LTS.

Reply:
That makes sense. I'll replace `lts/hydrogen` with `lts/*` in `.nvmrc`.

As you alluded to, the fallback `md4` implementation seems to be working. So, if we are OK with Webpack using an outdated hashing strategy, then I think we can proceed with these changes for now.

Reply:
I think that we are OK with `md4` for now because webpack will change the default hashing strategy to more suitable one eventually as you pointed out.